include/ofi_net.h

```diff
@@ -158,6 +158,57 @@ ofi_recvv_socket(SOCKET sock, struct iovec *iov, size_t cnt, int flags)
 
 ssize_t ofi_discard_socket(SOCKET sock, size_t len);
 
+/*
+ * Socket API
+ */
+struct ofi_sockapi {
+	ssize_t (*send)(struct ofi_sockapi *sockapi, SOCKET sock, const void *buf,
+			size_t len, int flags, void *ctx);
+	ssize_t (*sendv)(struct ofi_sockapi *sockapi, SOCKET sock,
+			 const struct iovec *iov, size_t cnt, int flags, void *ctx);
+	ssize_t (*recv)(struct ofi_sockapi *sockapi, SOCKET sock, void *buf,
+			size_t len, int flags, void *ctx);
+	ssize_t (*recvv)(struct ofi_sockapi *sockapi, SOCKET sock,
+			 struct iovec *iov, size_t cnt, int flags, void *ctx);
+};
```

Comment:
This is exactly along the lines of what I was thinking, though I was struggling to think of what to call it.

Reply:
Awesome!